runtime/compiler/runtime/CompileService.cpp

```diff
@@ -33,18 +33,26 @@ void J9CompileDispatcher::compile(JITServer::ServerStream *stream)
    TR::CompilationInfo * compInfo = getCompilationInfo(_jitConfig);
 
    TR_MethodToBeCompiled *entry = NULL;
-   if (TR::Options::getVerboseOption(TR_VerboseJITServer))
-      TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer, "Server received request for stream %p", stream);
+   if (!compInfo->getPersistentInfo()->getDisableFurtherCompilation())
```

Comment:
I think we need to set the `_disableFurtherCompilation` flag as volatile.

Reply:
Could you explain why is that?

Reply:
Flags that are used for synchronization (set by one thread and read by another) should generally be marked volatile to prevent the c++ compiler from caching the result in a register (thus updates to the flag become invisible). When a lot of code is involved, keeping a variable in a register is very unlikely because of register pressure, but better safe than sorry.